typescript/sdk/src/test/testUtils.ts

```diff
@@ -59,7 +59,9 @@ export function testCoreConfig(
     },
   };
 
-  return Object.fromEntries(chains.map((local) => [local, chainConfig]));
+  return chains.length
+    ? Object.fromEntries(chains.map((local) => [local, chainConfig]))
+    : chainConfig;
```

Comment:
Yeah this is kinda what I meant with my last comment. This pattern we have of configs types being like `config | string | chainmap<config>` leads to code being more complex than it needs to be. I'm not convinced the flexibility of passing in whatever is actually worth the trouble. I think we only did it in the past as a shortcut to get things working. @yorhodes @paulbalaji

Reply:
fixed.